docs/examples/viz_video_on_plane.py

```diff
@@ -0,0 +1,122 @@
+Plane, which can be extended to potentially all shapes
+"""
+
+from fury import window, actor
+from vtk.util import numpy_support
+import numpy as np
+import cv2
+import vtk
+
+import time
+
+
+# The Video Class
+# This Class wraps OpenCV Videocapture
+class Video:
+
+    def __init__(self, video, time):
+        self.path = video
+        self.video = cv2.VideoCapture(self.path)
+        self.fps = int(self.video.get(cv2.CAP_PROP_FPS))
+        self.frames = int(self.video.get(cv2.CAP_PROP_FRAME_COUNT))
+        self.time = time
+
+    # A generator to yield video frames on every call
+    def get_frame(self):
+        start = time.time()
+        for _ in range(self.frames):
+            isframe, frame = self.video.read()
+            dur = time.time() - start
+            print(dur)
+            if dur > self.time:
+                break
+            if isframe:
+                yield cv2.cvtColor(frame, cv2.COLOR_BGR2RGB)
+        self.video.release()
+        yield None
+
+
+class Animation:
+
+    def __init__(self, video, time=5, object="plane"):
+        # Initializes the Video object with the given Video
+        self.video = Video(video, time)
+        self.video_generator = self.video.get_frame()
+        self.current_video_frame = next(self.video_generator)
+        # Initialize Scene
+        self.initializeScene()
+        # Create a Show Manager and Initialize it
+        self.show_manager = window.ShowManager(self.scene,
+                                               size=(900, 768),
+                                               reset_camera=False,
+                                               order_transparent=True)
+        self.show_manager.initialize()
+
+    # Initialize the Scene with actors
+    def initializeScene(self):
+        self.scene = window.Scene()
+        # Initialize a Plane actor with the 1st video frame
+        self.setupPlaneActor()
+        self.scene.add(self.plane_actor)
+
+    def setupPlaneActor(self):
+        # set plane_actor with a texture from fury actor
+        self.plane_actor = actor.texture(self.current_video_frame)
+        # Create a vtkImageData object from the image and create a
+        # texture with it. Now, The texture object is assigned to
+        # the plane
+        self.image_shape = self.current_video_frame.shape
+        data = self.current_video_frame
+        self.grid = vtk.vtkImageData()
+        self.grid.SetDimensions(self.image_shape[1], self.image_shape[0], 1)
+        self.new_img_data = np.flip(data.swapaxes(0, 1), axis=1)\
+                              .reshape((-1, self.image_shape[-1]), order='F')
+        self.vtkarr = numpy_support.numpy_to_vtk(self.new_img_data, deep=False)
+        self.grid.GetPointData().SetScalars(self.vtkarr)
+        self.grid.GetPointData().Update()
+        self.texture = vtk.vtkTexture()
+        self.texture.SetInputDataObject(self.grid)
+        self.plane_actor.SetTexture(self.texture)
+
+    def changeImage(self):
```

Comment:
change_image

Reply:
This one should become and independent function too.

Reply:
The first one should be called ``initiate_video_texture`` and this one should be called ``update_video_texture``.